lib/helpers.js

```diff
@@ -1,17 +1,20 @@
 const _ = require('lodash');
 
-function resolveAuthLevel(schema, options, doc) {
+async function resolveAuthLevel(schema, options, doc) {
   // Look into options the options and try to find authLevels. Always prefer to take
   // authLevels from the direct authLevel option as opposed to the computed
   // ones from getAuthLevel in the schema object.
-  let authLevels = [];
+  let authLevelsIn = [];
   if (options) {
     if (options.authLevel) {
-      authLevels = _.castArray(options.authLevel);
+      authLevelsIn = options.authLevel;
     } else if (typeof schema.getAuthLevel === 'function') {
-      authLevels = _.castArray(schema.getAuthLevel(options.authPayload, doc));
```

Comment:
Why the approach of changes lines 7-17 as opposed to adding `await` just to this line?

Reply:
otherwise it's ambiguous what exactly is in the `authLevelsIn` variable. it could be an array, or it could be a promise.

Reply:
A few reasons, I'm casting it right after and throwing away authLevelsIn, adding `await Promise.resolve(...)` to each line is not super easy to read, and last but not least, the first iteration was using `.then()` so this would have been a lot more challenging.

Reply:
@makinde learned something new! await works on non-promise values